managed/resource_cluster.go

```diff
@@ -200,6 +193,284 @@ func (r resourceClusterType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Di
 				},
 			}),
 		},
+		"backup_replication_spec": {
+			Description: "Configuration for backup replication. Enables replication of cluster backups to offsite buckets.",
+			Optional:    true,
+			Computed:    true,
+			Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+				"gcp_spec": {
+					Description: "GCP-specific backup replication configuration.",
+					Optional:    true,
+					Computed:    true,
+					Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+						"enabled": {
+							Description: "Whether GCP backup replication is enabled for this cluster.",
+							Type:        types.BoolType,
+							Optional:    true,
+							Computed:    true,
+							PlanModifiers: []tfsdk.AttributePlanModifier{
+								tfsdk.UseStateForUnknown(),
+							},
+						},
+						"sync_cluster_spec": {
```

Comment:
Please check if you can use tfsdk validators here and for geo partitioned cluster spec, and reduce the custom validation logic. Keeps it cleaner I guess.